tests/integration/misc/usingUnboundModelsByPassingKnex.js

```diff
@@ -428,59 +429,59 @@ module.exports = session => {
             .eager(
               '[model1Relation1, model1Relation2.model2Relation1.[model1Relation1, model1Relation2]]'
             );
-        }).reflect(),
+        }).catch(err => createRejectionReflection(err)),
 
         Promise.try(() => {
           return Model1.query();
-        }).reflect(),
+        }).catch(err => createRejectionReflection(err)),
 
         Promise.try(() => {
           return Model1.query().where('id', 1);
-        }).reflect(),
+        }).catch(err => createRejectionReflection(err)),
 
         Promise.try(() => {
           return Model1.query().joinRelated('model1Relation1');
-        }).reflect(),
+        }).catch(err => createRejectionReflection(err)),
 
         Promise.try(() => {
           return Model1.query(session.knex)
             .findById(1)
             .then(model => {
               return model.$relatedQuery('model1Relation1');
             });
-        }).reflect(),
+        }).catch(err => createRejectionReflection(err)),
 
         Promise.try(() => {
           return Model1.query(session.knex)
             .findById(2)
             .then(model => {
               return model.$relatedQuery('model1Relation1Inverse');
             });
-        }).reflect(),
+        }).catch(err => createRejectionReflection(err)),
 
         Promise.try(() => {
           return Model1.query(session.knex)
             .findById(1)
             .then(model => {
               return model.$relatedQuery('model1Relation2');
             });
-        }).reflect(),
+        }).catch(err => createRejectionReflection(err)),
 
         Promise.try(() => {
           return Model2.query(session.knex)
             .findById(2)
             .then(model => {
               return model.$relatedQuery('model2Relation1');
             });
-        }).reflect(),
+        }).catch(err => createRejectionReflection(err)),
 
         Promise.try(() => {
           return Model1.query(session.knex)
             .findById(1)
             .then(model => {
               return model.$query();
             });
-        }).reflect()
+        }).catch(err => createRejectionReflection(err))
```

Comment:
This can be moved to map over Array<Promise> and 
`expect(promise).to.eventually.be.rejectedWith({name: '', message: ''})`

Reply:
Not forcing to this solution, just proposing

Reply:
Nah, you are right, that is a cleaner solution, current util method is basically a compatibility hack. Just need a bit of time to implement the change. Thanks for reviewing!